### ISSUE TITLE
high merror when using gblinear and softmax in large scale sparse one hot feature.

### ISSUE THREAD

**Issue by southisland** [dmlc/xgboost#3094]
Hi @tqchen 
Sorry for troubling you. 
We are using xgboost in large scale text multi-class classification. Each feature is an one-hot ngram. 
we use gblinear as booster and multi:softmax as objective.
8 million samples, 180 thousand features and 31 classes.
the train-merror is very high. I'm wondering if xgboost supports such scene. 
All suggestions are welcome and thanks a lot for your time. 

train and eval merror:
[1]	eval-merror:0.760152	train-merror:0.866071
[2]	eval-merror:0.842969	train-merror:0.879221
[3]	eval-merror:0.703361	train-merror:0.59508
[4]	eval-merror:0.703361	train-merror:0.59508
[5]	eval-merror:0.703361	train-merror:0.59508
[6]	eval-merror:0.703361	train-merror:0.59508
[7]	eval-merror:0.703361	train-merror:0.59508
[8]	eval-merror:0.703361	train-merror:0.59508
[9]	eval-merror:0.703361	train-merror:0.59508
[10]	eval-merror:0.703361	train-merror:0.59508
[11]	eval-merror:0.703361	train-merror:0.59508
[12]	eval-merror:0.703361	train-merror:0.59508
[13]	eval-merror:0.703361	train-merror:0.59508
[14]	eval-merror:0.703361	train-merror:0.59508
[15]	eval-merror:0.703361	train-merror:0.59508
[16]	eval-merror:0.703361	train-merror:0.59508
[17]	eval-merror:0.703361	train-merror:0.59508
[18]	eval-merror:0.703361	train-merror:0.59508
[19]	eval-merror:0.703361	train-merror:0.59508
[20]	eval-merror:0.703361	train-merror:0.59508
[21]	eval-merror:0.703361	train-merror:0.59508

**Comment by khotilov** [comment]
The regularizations parameters you are using might be rather non-optimal. Keep in mind that the scale of regularization parameters values here would be proportional to the number of rows (as opposed to how it is usually normalized WRT number of rows in regularized loss), which means that as the size of your sample grows, the optimal values of lambda and alpha would have to be increased accordingly (hint: optimize in a smaller sample, then scale to the full sample). Also, might try to lower eta and to run for longer.

**Comment by tqchen** [comment]
Thanks @khotilov for answering the question, new questions are welcomed at https://discuss.xgboost.ai/
